lib/pure/os.nim

```diff
@@ -296,6 +296,31 @@ proc setCurrentDir*(newDir: string) {.inline, tags: [].} =
   else:
     if chdir(newDir) != 0'i32: raiseOSError(osLastError())
 
+proc absolutePath*(path: string, root = getCurrentDir()): string =
+  ## Returns the absolute path of `path`, rooted at `root` (which must be absolute)
+  ## if `path` is absolute, return it, ignoring `root`
+  runnableExamples:
+    doAssert absolutePath("a") == getCurrentDir() / "a"
+  if isAbsolute(path): path
+  else:
+    doAssert root.isAbsolute, root
+    joinPath(root, path)
+
+when isMainModule:
+  # TODO: use doAssertRaises pending https://github.com/nim-lang/Nim/issues/8223
+  try:
+    let a = absolutePath("a", "b")
+    doAssert false, "should've thrown"
+  except:
```

Comment:
Technically, assertion failures are not catchable. You should raise an exception instead.

Reply:
actually that try/catch was buggy independently of the fact that assertion failures are not catchable; will fix; thanks!

Reply:
@dom96 just wrote https://github.com/nim-lang/Nim/issues/8237 in response to your comment

Reply:
> You should raise an exception instead

done